### PR TITLE
Feature/js-lingui

### PR DESCRIPTION
**Pull request by dhrubesh**
# Details

## Description

- [ ]  Remove all i18next dependencies from package.json

- [ ] Remove the i18next code from src/config/i18n.js

- [ ]  Setup JS-Lingui https://lingui.js.org/tutorials/react.html

- [ ] Using the CLI add the two languages that we support (en, fr)

- [ ] Migrate the existing translations in the components that we have

## Issue
#97

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +24,11 @@
     "sass": "sass --watch src/styles/scss:src/styles/css",
     "precommit": "lint-staged",
     "analyze": "source-map-explorer build/static/js/main.*",
-    "storybook": "start-storybook -p 9001 -c .storybook"
+    "storybook": "start-storybook -p 9001 -c .storybook",
+    "lingui-init": "lingui init",
```
Comment (crysfel) [inline package.json]: This task is not necessary, please remove it. The `lingui  init` should only be ran once when installing the project, you have done that already, no need for other to do it.

File: package.json [hunk 2]
```diff
@@ -26,7 +24,11 @@
...
+    "lingui-add-locale": "lingui add-locale en fr",
```
Comment (crysfel) [inline package.json]: Please remove this task, in the future we will use `lingui add-locale es jp` or any other locale we want, for now you have already installed en and fr, no need for others to run this task again.

File: src/components/contact/Contact.js [hunk 3]
```diff
@@ -24,14 +22,17 @@ const Contact = ({ t }) => {
         <img src={peopleSearch} alt="people search logo" />
       </div>
       <div className={styles.rightSection}>
-        <h1>{t('contact-page-header')}</h1>
+        <h1>
+          <Trans id="contact-page-header" />
+        </h1>
         <p>
-          <Interpolate i18nKey="contact-page-msg" slackOrg={slackOrg} email={emailLink} />
+          We want to hear your thoughts! Feel free to join our {slackOrg} or send us an email at{' '}
```
Comment (crysfel) [inline src/components/contact/Contact.js]: Let's use an id for this: `<Trans id="contact.msg">We want to hear your thoughts! Feel free to join our {slackOrg} or send us an email at</Trans>`

File: src/components/contact/Contact.js [hunk 4]
```diff
@@ -24,14 +22,17 @@ const Contact = ({ t }) => {
         <img src={peopleSearch} alt="people search logo" />
       </div>
       <div className={styles.rightSection}>
-        <h1>{t('contact-page-header')}</h1>
+        <h1>
+          <Trans id="contact-page-header" />
```
Comment (crysfel) [inline src/components/contact/Contact.js]: I think we should always set a default value: `<Trans id="contact.header" >Contact</Trans>`

File: src/components/contact/Contact.js [hunk 5]
```diff
@@ -1,20 +1,18 @@
...
+const Contact = () => {
   const emailLink = (
     <a target="_blank" href={`mailto:${process.env.REACT_APP_EMAIL}`} className={styles.email}>
       {process.env.REACT_APP_EMAIL}
     </a>
   );
   const slackOrg = (
     <a href={process.env.REACT_APP_SLACK_URL} className={styles.slackOrg}>
-      {t('slack-org')}
+      <Trans id="slack-org" />
```
Comment (crysfel) [inline src/components/contact/Contact.js]: Add the default text in english

File: src/components/navbar/Navbar.js [hunk 6]
```diff
@@ -1,22 +1,21 @@
...
 import styles from './navbar.scss';
+import { Trans, withI18n } from '@lingui/react';
 
-const Navbar = ({ t }) => (
+const Navbar = () => (
   <nav className={styles.navigationBar}>
     <div className={styles.leftNavLinks}>
       <a href="#" className={styles.navLinks}>
-        {t('About')}
+        <Trans id="About" />
```
Comment (crysfel) [inline src/components/navbar/Navbar.js]: for the id, let's use `[pageNameOrComponent].[uniqueKey]`, for example: `about.title`

```
<Trans id="about.title" >About</Trans>
```

File: src/components/navbar/Navbar.js [hunk 7]
```diff
@@ -1,22 +1,21 @@
...
       </a>
       <a href="#" className={styles.navLinks}>
-        {t('Mission')}
+        <Trans id="Mission" />
```
Comment (crysfel) [inline src/components/navbar/Navbar.js]: Same for this

File: src/containers/App/App.js [hunk 8]
```diff
@@ -1,20 +1,18 @@
 import React from 'react';
-import { func } from 'prop-types';
-import { translate } from 'react-i18next';
 import { Router } from '@reach/router';
 import { Home } from 'pages';
-import 'i18n/i18n';
+import { I18nProvider } from '@lingui/react';
+import catalogEn from '../../locale/en/messages.js';
+// import catalogFr from '../../locale/fr/messages.js';
```
Comment (crysfel) [inline src/containers/App/App.js]: Copy the current translation we already have to this file, and add it to the catalogs in line 10

File: src/locale/_build/src/components/contact/Contact.js.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (crysfel) [inline src/locale/_build/src/components/contact/Contact.js.json]: I'm wondering if we need the `_build` directory in git? I'm not really sure, can you investigate?

**Comment by tureey** [comment]
Once this PR is mmerged, would be okay to add a Documentation (I prefer to include the documentation in the same PR) to know how use it

**Comment by dhrubesh** [comment]
> Once this PR is mmerged, would be okay to add a Documentation (I prefer to include the documentation in the same PR) to know how use it

@crysfel @emmawedekind Can you guys create a new issue for Documentation?

**Comment by dhrubesh** [comment]
Hey @crysfel I've updated the PR in accordance with your comments. And about the `_build` folder, that gets generated during the **lingui-extract** (before the **lingui-compile** command) command itself. And if I remove that folder, JS-lingui isn't getting loading so I guess if we remove it, everyone will have to extract and compile JS-lingui manually.

**Comment by tureey** [comment]
I don't think what's best. For me, as far as we have documentation of how to extract... Is okay. But I don't know how good or bad practice is.

Talking about documentation.. I think we **we should include the documentation to our definition of done**. It means that if there is no documentation.. can't be merged.

**Comment by crysfel** [comment]
@dhrubesh ok, so let's remove it from git and compile it when building. Also... there's a webpack loader we can use to do the extracting for us: https://lingui.js.org/ref/loader.html

can you investigate how to use it? My guess is that we can add it to the `rules` here: https://github.com/Coding-Coach/coding-coach/blob/development/scripts/config-overrides.dev.js#L9

**Comment by crysfel** [comment]
Please rebase with latest in development

**Comment by crysfel** [comment]
@dhrubesh I've just fixed that problem, please rebase from the latest on develop and the linter will not throw an error anymore.

**Comment by crysfel** [comment]
What's the status on this? We should merge this PR soon, otherwise there will be conflicts everytime we merge something and all the new code needs to be migrated.

Let me know if you need any help, maybe someone else can take it from here?

**Comment by tureey** [comment]
same here. Ping me through slack if you need me! :fire:

**Comment by dhrubesh** [comment]
Hey I will appreciate if someone else can take it from here. I’ve got a tight schedule hence wont be getting time.

**Comment by tureey** [comment]
Is there something left? Only conflict solving? Or is there something else? @dhrubesh

**Comment by crysfel** [comment]
@dhrubesh That's cool, thanks for letting us know.

@tureey yes, conflicts and new merged code, do you want to take it? If not I can take it!

**Comment by tureey** [comment]
@crysfel If you have time, it's for you. I would have to use sleeping time if not. If you can't I will do it no problem. So don't feel bad if you don't have lots of time

**Comment by crysfel** [comment]
I will work on this!

**Comment by crysfel** [comment]
I'm going to open a new PR once #231 gets merged

**Pull request closed without merging**